### PR TITLE
Mappings for 1.18-pre1 placement modifiers

### PR DESCRIPTION
**Pull request by mschae23**
Decorators (now called "placement modifiers") have been refactored in 1.18 Pre-release 1.

### CODE REVIEW

File: mappings/net/minecraft/world/gen/decorator/PlacementModifiers.mapping [hunk 1]
```diff
@@ -0,0 +1,10 @@
+CLASS net/minecraft/class_6817 net/minecraft/world/gen/decorator/PlacementModifiers
+	FIELD field_36078 MOTION_BLOCKING_HEIGHTMAP Lnet/minecraft/class_6797;
+	FIELD field_36079 OCEAN_FLOOR_WG_HEIGHTMAP Lnet/minecraft/class_6797;
+	FIELD field_36080 WORLD_SURFACE_WG_HEIGTMAP Lnet/minecraft/class_6797;
+	FIELD field_36081 OCEAN_FLOOR_HEIGHTMAP Lnet/minecraft/class_6797;
+	FIELD field_36082 BOTTOM_TO_TOP_RANGE Lnet/minecraft/class_6797;
+	FIELD field_36086 BOTTOM_TO_120 Lnet/minecraft/class_6797;
```
Comment (enbrain) [inline mappings/net/minecraft/world/gen/decorator/PlacementModifiers.mapping]:
```suggestion
	FIELD field_36086 BOTTOM_TO_120_RANGE Lnet/minecraft/class_6797;
```
For consistency.

File: mappings/net/minecraft/world/gen/decorator/AbstractConditionalPlacementModifier.mapping [hunk 2]
```diff
@@ -1,4 +1,5 @@
-CLASS net/minecraft/class_6661 net/minecraft/world/gen/decorator/ConditionalDecorator
+CLASS net/minecraft/class_6661 net/minecraft/world/gen/decorator/AbstractConditionalPlacementModifier
```
Comment (enbrain) [inline mappings/net/minecraft/world/gen/decorator/AbstractConditionalPlacementModifier.mapping:1]: Why not just `ConditionalPlacementModifier`?

File: mappings/net/minecraft/world/gen/decorator/PlacementModifiers.mapping [hunk 3]
```diff
@@ -0,0 +1,10 @@
+CLASS net/minecraft/class_6817 net/minecraft/world/gen/decorator/PlacementModifiers
+	FIELD field_36078 MOTION_BLOCKING_HEIGHTMAP Lnet/minecraft/class_6797;
+	FIELD field_36079 OCEAN_FLOOR_WG_HEIGHTMAP Lnet/minecraft/class_6797;
+	FIELD field_36080 WORLD_SURFACE_WG_HEIGTMAP Lnet/minecraft/class_6797;
```
Comment (Shnupbups) [inline mappings/net/minecraft/world/gen/decorator/PlacementModifiers.mapping]:
```suggestion
	FIELD field_36080 WORLD_SURFACE_WG_HEIGHTMAP Lnet/minecraft/class_6797;
```

File: mappings/net/minecraft/world/gen/decorator/PlacementModifiers.mapping [hunk 4]
```diff
@@ -0,0 +1,10 @@
+CLASS net/minecraft/class_6817 net/minecraft/world/gen/decorator/PlacementModifiers
```
Comment (Shnupbups) [inline mappings/net/minecraft/world/gen/decorator/PlacementModifiers.mapping]: Should be called `PlacedFeatures`, as it contains the registration method for `PlacedFeature`s as well as the method for getting a default one for the defaulted registry.

File: mappings/net/minecraft/world/gen/decorator/RandomOffsetPlacementModifier.mapping [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+	FIELD field_35421 spreadY Lnet/minecraft/class_6017;
+	METHOD <init> (Lnet/minecraft/class_6017;Lnet/minecraft/class_6017;)V
+		ARG 1 xzSpread
+		ARG 2 ySpread
+	METHOD method_39656 vertically (Lnet/minecraft/class_6017;)Lnet/minecraft/class_6732;
+		ARG 0 spreadY
+	METHOD method_39657 of (Lnet/minecraft/class_6017;Lnet/minecraft/class_6017;)Lnet/minecraft/class_6732;
+		ARG 0 spreadXz
+		ARG 1 spreadY
+	METHOD method_39658 hroizontally (Lnet/minecraft/class_6017;)Lnet/minecraft/class_6732;
```
Comment (Shnupbups) [inline mappings/net/minecraft/world/gen/decorator/RandomOffsetPlacementModifier.mapping]:
```suggestion
	METHOD method_39658 horizontally (Lnet/minecraft/class_6017;)Lnet/minecraft/class_6732;
```

**Review by Shnupbups — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/world/gen/decorator/AbstractConditionalPlacementModifier.mapping (hunk 2 above)
Comment (mschae23) [reply mappings/net/minecraft/world/gen/decorator/AbstractConditionalPlacementModifier.mapping:1]: The `Abstract` prefix would be consistent with `AbstractCountPlacementModifier`, and shows that it isn't a placement modifier itself, but is supposed to be used by other modifiers.

**Review by Shnupbups — APPROVED** [review]

**Review by enbrain — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]
👍

**Pull request merged**